### ISSUE TITLE
Umbrella app with mix releases

### ISSUE THREAD

**Issue by agramichael** [Logflare/logflare_logger_backend#42]
Can't get the logger backend working in our umbrella app using mix releases on Gigalixir.
The log drain works fine but we want to add custom metadata to our logs.

The mix command says the logger is configured properly but when running a release nothing is sent.

**Comment by chasers** [comment]
Do you have it in all your mix files?  I needs to be in all your mix files or the app won't get started for each app.  This was the solution for the last person who had an issue with an umbrella app setup.

**Comment by agramichael** [comment]
Added to all mix files and configuring the `:logflare_logger_backend` in the umbrella `config.exs`.

Getting an error when starting a release (step that runs migrations): `Logflare source_id for LogflareLogger backend is NOT configured` 

`mix test` works as expected and `:source_id` is in my env or else `fetch_env!` in `releases.exs` would raise an error.

**Comment by agramichael** [comment]
I'm not entirely sure but I think the library only checks for config in the Application environment and when `eval` is run it is empty. Maybe there should be some second try which checks the system environment, like what sentry-elixir uses?

**Comment by chasers** [comment]
> Getting an error when starting a release (step that runs migrations)

Which migrations exactly?  Ecto migrations?

> Can confirm the release start works but running migrations via an eval before starting raises the error.

Can you expand on this a bit? If it works with release `start` what are you doing exactly in the migrations?

**Comment by chasers** [comment]
And if you can post any code that will surely help.  If you'd like to send over something sensitive you can email it to `support@logflare.app`.

**Comment by agramichael** [comment]
Basically, before we run `start`, we run an `eval "MyApp.Release.migrate"`. And, apparently, when `eval` is run, the application environment for the `logflare_logger_backend` is empty, which causes the error.

I am not sure if there is a way to force the config to be loaded when running `eval`.

**Comment by agramichael** [comment]
Made a sample app so you can test this out.

https://github.com/agramichael/logflare_umbrella_test

After setting up the db and deps, build a release and run `eval "MyApp.Release.migrate"`

**Comment by chasers** [comment]
Okay this is perfect thank you. Will follow up.

**Comment by chasers** [comment]
Ah this is not an umbrella though, and the configs are in config.exs no releases.exs. Actually looks like we replicated your issue yesterday, give us a couple days here. Thanks!

**Comment by agramichael** [comment]
Yeah I realised this isn't specific to umbrella apps or the task I was trying to accomplish, so I made the sample app as plain as possible. Thanks for looking into this!

**Comment by agramichael** [comment]
@chasers any updates on this?

**Comment by chasers** [comment]
Yep we're close. Thanks for your patience!

**Comment by chasers** [comment]
@agramichael okay two options...

## Upgrade to Elixir 1.11

What are the chances you can upgrade to Elixir v1.11?

```
When I put the runtime config into runtime.exs , introduced in Elixir 1.11, instead of releases.exs , everything works as expected without any other modifications
```

btw, releases.exs is now deprecated

## Attach the LogflareLogger backend dynamically

Add `Logger.add_backend(LogflareLogger.HttpBackend)` to your `application.ex` right after def start. Configs should all work as expected.

An issue here is that you don't get application logging during boot. 

Another issue is that it won't get restarted dynamically as it's not in the logger supervision tree.

Here is the road we went down...

```
so there are multiple aspects at play here
8:04
first of all, for whatever reason, releases.exs configs are not properly loaded into application when running ../app_release eval
8:05
eval command, as the rpc comands the start_clean executable that boots only kernel without starting the actual release application supervision tree
8:06
however, since elixir 1.10 release, erlang logger is integrated with elixir logger
8:06
and erl logger is in erlang kernel since OTP 21.0
8:07
so when the start_clean command was invoked it started the erlang logger which in turn started the elixir logger, which attempted to attach the configured backends
8:07
I've checked that it happens even if logger is not present in extra_application in mix.exs
8:08
if eval load the releases.exs config values as expected, that would be fine but it doesn't
8:09
so one of the solutions was to put this line export RELEASE_BOOT_SCRIPT_CLEAN="start" into env.sh.eex
8:10
the default value for RELEASE_BOOT_SCRIPT_CLEAN shell env variable which eval uses is start_clean
8:10
this worked on a minimal project with no repos or other stuff
8:11
but starting the app supervision tree during the eval for a complex app would most probably lead to errors
```

**Comment by agramichael** [comment]
@chasers Thanks for investigating this! Will try with Elixir 1.11 and `runtime.exs` and get back to you.

**Comment by agramichael** [comment]
@chasers Can confirm this solved the issue!
